### ISSUE TITLE
about the vocab.pkl for smile transformer

### ISSUE THREAD

**Issue by fuxuliu** [HanselYu/PreKcat#3]
Hi, thanks for you great work.
I want to use your model to do some inference, but I did not find the vocab.pkl file from smile transformer repo, would you provide this file or upload this file to Prekcat repo.

Thanks.

**Comment by HanselYu** [comment]
I have added this file to the directory. I hope it could help you.

**Comment by fuxuliu** [comment]
Thanks for your vocab.pkl file.
I am trying to reproduce the performance of the PreKcat paper based on the code of eature extraction on the readme, but I can't get the rmse or R mentioned in the PreKcat paper.
The code is going to be a little bit different here is the model fetch for prot_t5, but I think there should be no difference.
```
## readme version
tokenizer = T5Tokenizer.from_pretrained("prot_t5_xl_uniref50", do_lower_case=False)
model = T5EncoderModel.from_pretrained("prot_t5_xl_uniref50")
## new version
model = T5EncoderModel.from_pretrained("Rostlab/prot_t5_xl_uniref50")
tokenizer = T5Tokenizer.from_pretrained("Rostlab/prot_t5_xl_uniref50", do_lower_case=False)
```

I used dlkcat data and filtered it, leaving 16,838 samples, just like the PreKcat paper mentioned. Then I use the complete data for inference, and the result obtained should be better than the performance in the paper, because it includes the training set.
But maybe there's something wrong with my labeling operation.

May I ask if it is possible for you to provide a csv file for your training, which contains protein sequence, smile sequence, kcat, log kcat, and train/valid/test split for your experiment?
And provides an inference script that replicates the performance of test set.

Really looking forward to your help, thank you

**Comment by HanselYu** [comment]
How do you handle labels? Here, I referred to the DLKcat literature and took the logarithm of the labels with base 10. The other steps should be the same.

**Comment by fuxuliu** [comment]
okay, I had a question here.
Did you calculate r2 score and rmse by using label after log10?
e.g.
`
preds = model(inputs)

rmse = np.sqrt(mse(preds, log10_kcat_label))

r2 = r2score(preds, log10_kcat_label)
`

**Comment by HanselYu** [comment]
That's right.

**Comment by fuxuliu** [comment]
Thanks, I think I got the right results.
I try to retrain a NN model with the feature as the this repo method.
I finish some  experiments, the tree model is outperform the NN model, such as the simple Linear model.
Did you have the similar conclusion? because the PreKcat use the tree model.

**Comment by HanselYu** [comment]
To be honest, I have indeed tried some common ensemble models, including Random Forest, XGBoost and so on. The tree-based models perform well, but I haven't conducted a systematic comparative analysis yet. I plan to do some model and parameter optimization in the future. But now, we are conducting more experiments involving wet-lab test.
